### PR TITLE
Supporting data label for hyperopt

### PR DESCRIPTION
**Pull request by martin1tab**
Add the possibility to run hyperopt on multiple data sets in the same ramp kit by specifying `--data-label`

### CODE REVIEW

File: rampwf/hyperopt/tests/test_hyperparameter.py [hunk 1]
```diff
@@ -36,10 +36,25 @@ def test_hyperparameter():
 
 
 @pytest.mark.parametrize("submission", ['starting_kit', 'one_hyper_kit'])
-def test_hyperopt(submission):
+def test_hyperopt_data_label_notNone(submission):
```
[the remark this answers is not in the document]
Comment (albertcthomas) [reply rampwf/hyperopt/tests/test_hyperparameter.py:39]: actually maybe a better name would be `test_hyperopt_without_data_label`

File: rampwf/hyperopt/tests/test_hyperparameter.py [hunk 2]
```diff
@@ -36,10 +36,25 @@ def test_hyperparameter():
...
     run_hyperopt(
         ramp_kit_dir, ramp_kit_dir,
-        os.path.join(ramp_kit_dir, 'submissions'),
-        submission, 'random', 64, is_cleanup=True)
+        os.path.join(ramp_kit_dir, 'submissions'), 'titanic',
+        submission, 'random', 64, True)
+
+
+@pytest.mark.parametrize("submission", ['starting_kit', 'one_hyper_kit'])
+def test_hyperopt_data_label_None(submission):
```
Comment (albertcthomas) [inline rampwf/hyperopt/tests/test_hyperparameter.py:54]: `None`: same remark as above

File: rampwf/hyperopt/tests/test_hyperparameter.py [hunk 3]
```diff
@@ -36,10 +36,25 @@ def test_hyperparameter():
...
 @pytest.mark.parametrize("submission", ['starting_kit', 'one_hyper_kit'])
-def test_hyperopt(submission):
+def test_hyperopt_data_label_notNone(submission):
     ramp_kit_dir = os.path.join(
-        PATH, 'interfaces', 'header_in_files', 'titanic')
+        PATH, 'interfaces', 'header_in_files', 'classifier_kit')
+    destination_folder = \
+        os.path.join(ramp_kit_dir, 'submissions',
+                     'one_hyper_kit_titanic_hyperopt')
+    if not os.path.exists(destination_folder):
```
Comment (albertcthomas) [inline rampwf/hyperopt/tests/test_hyperparameter.py:45]: why do you need this?

Comment (martin1tab) [reply rampwf/hyperopt/tests/test_hyperparameter.py:45]: for test coverage (to test a if/else branch)

**Review by albertcthomas — APPROVED** [review]
thanks @martin1tab

**Comment by kegl** [comment]
Awesome, thank you @martin1tab .

@agramfort if no objection from the Saclay side, we'll merge this next week. It only touches hyperopt so no core functions that you use.

**Review by agramfort — COMMENTED** [review]
no objection if it does not break any test here

**Pull request merged**